velox/common/encode/Base64.cpp

```diff
@@ -104,26 +117,40 @@ static_assert(
         kBase64UrlCharset,
         kBase64UrlReverseIndexTable),
     "kBase64UrlCharset has incorrect entries");
-// Similar to strchr(), but for null-terminated const strings.
-// Another difference is that we do not consider "\0" to be present in the
-// string.
-// Returns true if "str" contains the character c.
-constexpr bool constCharsetContains(
+
+/// Searches for a character within a charset up to a certain index.
+static constexpr bool findCharacterInCharSet(
     const Base64::Charset& charset,
     uint8_t idx,
     const char c) {
-  return idx < charset.size() &&
-      ((charset[idx] == c) || constCharsetContains(charset, idx + 1, c));
+  for (uint8_t i = 0; i <= idx; ++i) {
```

Comment:
Is this equivalent to the recursive implementation?
The recursive goes from idx until `idx < charset.size()`?
But the iterative goes from 0 to idx?

Reply:
Please add a test that covers this.

Reply:
updated the code